src/pgduckdb_hooks.cpp

```diff
@@ -350,6 +350,8 @@ DuckdbExplainOneQueryHook(Query *query, int cursorOptions, IntoClause *into, Exp
 	 * standard_ExplainOneQuery).
 	 */
 	duckdb_explain_analyze = es->analyze;
+	if(es->format == EXPLAIN_FORMAT_JSON)
+		duckdb_explain_format = duckdb::ExplainFormat::JSON;
```

Comment:
this also needs an `else`, because now once you did one explain with `JSON` it's never reset anymore. But probably easiest way is to store the original Postgres value (i.e. `EXPLAIN_FORMAT_JSON`) always in this global, instead of the duckdb explain format.